src/test/java/com/bookstore/service/book/BookServiceImplTest.java

```diff
@@ -0,0 +1,147 @@
+    void findAll_validPageable_returnsValidListBookResponseDtos() {
+        when(bookRepository.findAll(PAGEABLE))
+                .thenReturn(new PageImpl<>(List.of(VALID_BOOK)));
+        when(bookMapper.toDto(VALID_BOOK)).thenReturn(VALID_BOOK_RESPONSE_DTO);
+
+        List<BookResponseDto> expected = List.of(VALID_BOOK_RESPONSE_DTO);
+
+        assertEquals(expected, bookServiceImpl.findAll(PAGEABLE));
+    }
+
+    @Test
+    @DisplayName("Valid case of the getById method run")
+    void getById_validBookID_returnsValidBookResponseDto() {
+        when(bookRepository.findById(VALID_ID)).thenReturn(Optional.ofNullable(VALID_BOOK));
+        when(bookMapper.toDto(VALID_BOOK)).thenReturn(VALID_BOOK_RESPONSE_DTO);
+
+        assertEquals(VALID_BOOK_RESPONSE_DTO, bookServiceImpl.getById(VALID_ID));
+    }
+
+    @Test
+    @DisplayName("Valid case of the save method run")
+    void save_validBookRequestDto_returnsValidBookResponseDto() {
+        when(bookRepository.save(VALID_BOOK)).thenReturn(VALID_BOOK);
+        when(bookMapper.toModel(VALID_BOOK_REQUEST_DTO)).thenReturn(VALID_BOOK);
+        when(bookMapper.toDto(VALID_BOOK)).thenReturn(VALID_BOOK_RESPONSE_DTO);
+
+        assertEquals(VALID_BOOK_RESPONSE_DTO, bookServiceImpl.save(VALID_BOOK_REQUEST_DTO));
+    }
+
+    @Test
+    @DisplayName("Valid case of the update method run")
+    void update_validBookRequestDtoAndID_returnsValidBookResponseDto() {
+        when(bookRepository.save(VALID_BOOK)).thenReturn(VALID_BOOK);
+        when(bookRepository.findById(VALID_ID)).thenReturn(Optional.of(VALID_BOOK));
+        when(bookMapper.toDto(VALID_BOOK)).thenReturn(VALID_BOOK_RESPONSE_DTO);
+
+        assertEquals(VALID_BOOK_RESPONSE_DTO, bookServiceImpl
+                .update(VALID_ID, VALID_BOOK_REQUEST_DTO));
+    }
+
+    @Test
+    @DisplayName("Valid case of the deleteById method run")
+    void deleteById_validBookID_isOk() {
+        doNothing().when(bookRepository).deleteById(VALID_ID);
+        bookServiceImpl.deleteById(VALID_ID);
+        verify(bookRepository, times(1)).deleteById(VALID_ID);
+    }
+
+    @Test
+    @DisplayName("Valid case of the findAllByCategoryId method run")
+    void findAllByCategoryId_validPageableAndCategoryID_returnsValidListDtos() {
+        when(bookMapper.toDtoWithoutCategories(VALID_BOOK))
+                .thenReturn(VALID_BOOK_DTO_WITHOUT_CATEGORY_IDS);
+        when(bookRepository.findAllByCategoryId(PAGEABLE, VALID_ID))
+                .thenReturn(List.of(VALID_BOOK));
+
+        List<BookResponseDtoWithoutCategoryIds> expected =
+                List.of(VALID_BOOK_DTO_WITHOUT_CATEGORY_IDS);
+
+        assertEquals(expected, bookServiceImpl.findAllByCategoryId(PAGEABLE, VALID_ID));
+    }
+
+    @Test
+    @DisplayName("Valid case of the getBookById method run")
+    void getBookById_validBookID_returnsValidBook() {
+        when(bookRepository.findById(VALID_ID)).thenReturn(Optional.of(VALID_BOOK));
+
```

Comment:
Please make a call to the system under test explicitly, it is not clear when you call 'bookServiceImpl' in assert block